src/routes/search.js

```diff
@@ -101,46 +113,92 @@ class SearchRoute extends Component {
+   * @returns {String} the final url for a search type, including a query
+   */
+  buildSearchUrl(pathname, query = '', searchType = this.state.searchType) {
+    let queryString = typeof query === 'string' ? query : qs.stringify(query);
+    let url = `${pathname}?searchType=${searchType}`;
+
+    if (queryString) {
+      url += `&${queryString}`;
+    }
 
-    this.performSearch({ ...query, search });
-  };
+    return url;
+  }
 
-  render() {
-    const { match: { params: { searchType } }, search } = this.props;
-    const { [searchType]: { query, isResolved, isPending, isRejected } } = search;
-
-    const searchTypeLocations = Object.keys(search).reduce((locations, type) => ({
-      ...locations,
-      [type]: {
-        pathname: `/eholdings/search/${type}`,
-        search: queryString.stringify(search[type].query)
-      }
+  /**
+   * Generates a hash of urls for the various search types and their queries
+   * @returns {Object} key value pair of search type urls
+   */
+  getSearchTypeUrls() {
+    let { location, search } = this.props;
+
+    return ['vendors', 'packages', 'titles'].reduce((locations, type) => ({
+      [type]: this.buildSearchUrl(location.pathname, search[type].query, type),
+      ...locations
     }), {});
+  }
 
-    return (
-      <div data-test-eholdings>
-        <SearchPaneset
-          hideFilters={!!this.props.location.search}
-          hasResults={isPending || isResolved || isRejected}
-          resultsType={searchType}
-          searchForm={(
-            <SearchForm
-              searchType={searchType}
-              searchString={query.search}
-              searchTypeLocations={searchTypeLocations}
-              onSearch={this.handleSearch}
-            />
-          )}
-        >
-          {this.props.children}
-        </SearchPaneset>
-      </div>
-    );
+  /**
+   * Renders the search results component specific to the current search type
+   */
+  renderResults() {
+    let { location } = this.props;
+    let { searchType, search } = this.state;
+    let { isResolved, isPending, isRejected } = search;
+
+    if (isResolved || isPending || isRejected) {
+      if (searchType === 'vendors') {
+        return <VendorSearchResultsRoute location={location} />;
+      } else if (searchType === 'packages') {
+        return <PackageSearchResultsRoute location={location} />;
+      } else if (searchType === 'titles') {
+        return <TitleSearchResultsRoute location={location} />;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * If there is a valid search type present in the current location,
+   * render the search paneset, otherwise simply render our children
+   */
+  render() {
+    let { location, children } = this.props;
+    let { searchType, search } = this.state;
+
+    if (searchType) {
+      let hideDetails = /^\/eholdings\/?$/.test(location.pathname);
+      let { query } = search;
+
+      return (
+        <div data-test-eholdings>
+          <SearchPaneset
```

Comment:
It might even make sense to delegate to a search view at this point. and put the paneset in that

Reply:
That's what this is. The only thing that this route renders is the search paneset with the search form. I suppose we could just call it `searchView` and include the form inside of that component. But for this route that paneset is essentially the only view.

Reply:
The main reason I suggest it is that `renderResults` also feels like a component, or "view-y"

Reply:
True. It used to be just a simple ternary in the render method, but I tried to pulled out most search-type specific things into their own methods (besides lines 97-99).

Essentially, it was this:
``` javascript
  resultsView={(isResolved || isPending || isRejected) && ResultsComponent}
```

It could also be something like `getResultsComponent` which would return the component constructor and the render method can give it props

Reply:
I guess we could argue that this route is responsible for _multiple_ views. Search, Results, and Details.